examples/never-type/never_type.md

```diff
@@ -0,0 +1,7 @@
+# Never type
+
+`never` means that no value belongs to the type.
```

Comment:
Why have we omitted content that is available in the slides? We can also update the bal file to include an example for `xml<never>`.

Reply:
Added an example